### PR TITLE
(SIMP-6787) Documentation Updates

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Cleaned up for `puppet strings`
* Added REFERENCE.md
* Updated README.md
* Switched FIPS check to system-only
* Added Puppet 6 support

SIMP-6787 #close

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -181,35 +148,44 @@ pup5-unit:
...
+  <<: *pup_5_5_10
   <<: *acceptance_base
   script:
-    - 'bundle exec rake spec_clean'
-    - 'bundle exec rake beaker:suites[default]'
+    - 'bundle exec rake beaker:suites'
 
-pup4.10:
-  <<: *pup_4_10
+pup5.5.10-fips:
```
Comment (lnemsick-simp) [inline .gitlab-ci.yml]: This should not have been added.  See line 201 of the original .gitlab-ci.yml file 'Note:: Radius does not work in fips mode'

File: README.md [hunk 2]
```diff
@@ -1,81 +1,99 @@
...
-using LDAP.
+This basic setup will configure RADIUS to listen on all interfaces and
+authenticate using LDAP.
 
 #### Install freeradius and the LDAP module and site configuration.
 
-Add the following to hiera for the Radius server:
+Add the following to Hiera for the Radius server:
 
 ```yaml
```
Comment (lnemsick-simp) [inline README.md]: In another PR you called out how the `classes` key was for a SIMP system.  Shouldn't you do that here, as well?

File: README.md [hunk 3]
```diff
@@ -128,95 +147,82 @@ or to copy over a file with clients defined, set the hiera variable:
...
+  goes here
 ```
 
 
 ### Other configuration
 
-The following configurations are not needed for the LDAP configuration.  These are a few
-examples of further configuration or alternate configuration that can be done.
-
-#### Add local radius users and trigger.
```
Comment (lnemsick-simp) [inline README.md]: Why was this example removed?

File: manifests/v3/sites/ldap.pp [hunk 4]
```diff
@@ -28,21 +30,20 @@
...
+  String               $group            = simplib::lookup( 'freeradius::group', {'default_value' => 'radiusd'} ),
+  Integer              $max_connections  = 16,
+  Integer              $lifetime         = 0,
+  Integer              $idle_timeout     = 30
 ){
 
   include 'freeradius'
 
-  if $fips or $facts['fips_enabled'] {
+  if $facts['fips_enabled'] {
```
Comment (lnemsick-simp) [inline manifests/v3/sites/ldap.pp]: IMHO the original implementation was correct.  That way the user is notified immediately that they have bad configuration.  As this change is written, they will find out after a system reboot once FIPS is applied.

File: README.md (hunk 3 above)
Comment (trevor-vaughan) [reply README.md]: @lnemsick-simp The example parameters were incorrect and I don't know the correct content to add to make this work properly. You have to know more about the `freeradius` configuration than a starting user would understand to make any advanced configuration work but it can be inferred from the rest of the documentation.

File: manifests/v3/modules/ldap.pp [hunk 5]
```diff
@@ -74,74 +70,67 @@
...
+  Optional[String]            $profile_attribute                              = undef,
+  Stdlib::AbsolutePath        $random_file                                    = '/dev/urandom',
+  String                      $require_cert                                   = 'demand',
+  Integer[1]                  $retry_delay                                    = 30,
+  Boolean                     $start_tls                                      = true,
+  String                      $user_filter                                    = '(uid=%{%{Stripped-User-Name}:-%{User-Name}})',
+  Optional[String]            $user_access_attribute                          = undef,
+  Boolean                     $user_access_positive                           = true,
+  Optional[Freeradius::Scope] $user_scope                                     = undef
+) inherits freeradius {
```
Comment (lnemsick-simp) [inline manifests/v3/modules/ldap.pp]: Why did you remove the fips check.  Please put it back.

Comment (lnemsick-simp) [reply manifests/v3/modules/ldap.pp]: Didn't see inheritance.  Disregard.

**Comment by lnemsick-simp** [comment]
https://gitlab.com/simp/pupmod-simp-freeradius/pipelines/70338492

**Comment by lnemsick-simp** [comment]
https://gitlab.com/simp/pupmod-simp-freeradius/pipelines/70345355
